### ISSUE TITLE
Event log messages show the wrong description

### ISSUE THREAD

**Issue by mtaanquist** [microsoft/navcontainerhelper#2581]
**Describe the issue**
When attempting to inspect the Event Logs from a container, they all have the wrong description in the "General" tab of a log entry, while the XML shows the correct details.

**Scripts used to create container and cause the issue**
```
New-BcContainer bctest200 `
    -accept_eula `
    -alwaysPull `
    -artifactUrl (Get-BCArtifactUrl -type OnPrem -country DK -version 20) `
    -assignPremiumPlan `
    -auth NavUserPassword `
    -doNotExportObjectsToText `
    -enableTaskScheduler `
    -isolation process `
    -licenseFile "BC 20 DEV.flf" `
    -updateHosts 
```

**Full output of scripts**
```
BcContainerHelper version 3.0.11
BcContainerHelper emits usage statistics telemetry to Microsoft
BcContainerHelper is version 3.0.11
BcContainerHelper is running as administrator
Hyper-V is Enabled
UsePsSession is True
Host is Microsoft Windows 10 Enterprise - 21H2
Docker Client Version is 20.10.16
Docker Server Version is 20.10.16
Removing Session bctest200
Removing container bctest200
Removing entries from hosts
Removing bctest200 from container hosts file
Removing bctest200-* from container hosts file
Removing Desktop shortcuts
Removing C:\ProgramData\BcContainerHelper\Extensions\bctest200
Fetching all docker images
Fetching all docker volumes
INFO: Windows 10 21H1/21H2 images are not yet available, using 2004 as these are found to work better than 20H2 on 21H1/21H2
Pulling image mcr.microsoft.com/businesscentral:10.0.19041.1415
10.0.19041.1415: Pulling from businesscentral
Digest: sha256:aa9f9d0779654ab9dba736a43141591a70fa419c4a9a57fd8a517086841280ff
Status: Image is up to date for mcr.microsoft.com/businesscentral:10.0.19041.1415
mcr.microsoft.com/businesscentral:10.0.19041.1415
Using image mcr.microsoft.com/businesscentral:10.0.19041.1415
Creating Container bctest200
Style: onprem
Multitenant: No
Version: 20.2.41144.41423
Platform: 20.0.41086.41354
Generic Tag: 1.0.2.10
Container OS Version: 10.0.19041.1415 (2004)
Host OS Version: 10.0.19044.1766 (21H2)
Using process isolation
Using locale da-DK
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Using license file C:\BC 20 DEV.flf
Additional Parameters:
--env customNavSettings=EnableTaskScheduler=True
Files in C:\ProgramData\BcContainerHelper\Extensions\bctest200\my:
- AdditionalOutput.ps1
- license.flf
- MainLoop.ps1
- SetupNavUsers.ps1
- SetupVariables.ps1
- updatehosts.ps1
Creating container bctest200 from image mcr.microsoft.com/businesscentral:10.0.19041.1415
eb37539c44002f5150276135476b9be6e00f4fe7ab6ad151e62986a0beea035e
Waiting for container bctest200 to be ready
Using artifactUrl https://bcartifacts.azureedge.net/onprem/20.2.41144.41423/dk
Using installer from C:\Run\150-new
Installing Business Central
Installing from artifacts
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
c:\dl\onprem\20.2.41144.41423\platform\ServiceTier\Program Files
c:\dl\onprem\20.2.41144.41423\platform\ServiceTier\System64Folder
Copying PowerShell Scripts
c:\dl\onprem\20.2.41144.41423\platform\WindowsPowerShellScripts\Cloud\NAVAdministration
c:\dl\onprem\20.2.41144.41423\platform\WindowsPowerShellScripts\WebSearch
Copying Web Client Files
c:\dl\onprem\20.2.41144.41423\platform\WebClient\Microsoft Dynamics NAV
Copying Client Files
c:\dl\onprem\20.2.41144.41423\platform\LegacyDlls\program files\Microsoft Dynamics NAV
c:\dl\onprem\20.2.41144.41423\platform\LegacyDlls\program files\Microsoft Dynamics NAV
c:\dl\onprem\20.2.41144.41423\platform\LegacyDlls\systemFolder
Copying ModernDev Files
c:\dl\onprem\20.2.41144.41423\platform
c:\dl\onprem\20.2.41144.41423\platform\ModernDev\program files\Microsoft Dynamics NAV
Copying additional files
Copying ConfigurationPackages
C:\dl\onprem\20.2.41144.41423\dk\ConfigurationPackages
Copying Test Assemblies
C:\dl\onprem\20.2.41144.41423\platform\Test Assemblies
Copying Applications
C:\dl\onprem\20.2.41144.41423\dk\Applications
Copying dependencies
Copying ReportBuilder
Using artifactUrl https://bcartifacts.azureedge.net/onprem/20.2.41144.41423/dk
Using installer from
Installing Business Central
Installing from artifacts
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
c:\dl\onprem\20.2.41144.41423\platform\ServiceTier\Program Files
c:\dl\onprem\20.2.41144.41423\platform\ServiceTier\System64Folder
Copying PowerShell Scripts
c:\dl\onprem\20.2.41144.41423\platform\WindowsPowerShellScripts\Cloud\NAVAdministration
c:\dl\onprem\20.2.41144.41423\platform\WindowsPowerShellScripts\WebSearch
Copying Web Client Files
c:\dl\onprem\20.2.41144.41423\platform\WebClient\Microsoft Dynamics NAV
Copying Client Files
c:\dl\onprem\20.2.41144.41423\platform\LegacyDlls\program files\Microsoft Dynamics NAV
c:\dl\onprem\20.2.41144.41423\platform\LegacyDlls\program files\Microsoft Dynamics NAV
c:\dl\onprem\20.2.41144.41423\platform\LegacyDlls\systemFolder
Copying ModernDev Files
c:\dl\onprem\20.2.41144.41423\platform
c:\dl\onprem\20.2.41144.41423\platform\ModernDev\program files\Microsoft Dynamics NAV
Copying additional files
Copying ConfigurationPackages
C:\dl\onprem\20.2.41144.41423\dk\ConfigurationPackages
Copying Test Assemblies
C:\dl\onprem\20.2.41144.41423\platform\Test Assemblies
Copying Applications
C:\dl\onprem\20.2.41144.41423\dk\Applications
Copying dependencies
Copying ReportBuilder
Importing PowerShell Modules
Restoring CRONUS Demo Database
Setting CompatibilityLevel for CRONUS on localhost\SQLEXPRESS
Modifying Business Central Service Tier Config File for Docker
Creating Business Central Service Tier
Installing SIP crypto provider: 'C:\Windows\System32\NavSip.dll'
Starting Business Central Service Tier
Importing license file
Stopping Business Central Service Tier
Installation took 162 seconds
Installation complete
Initializing...
Setting host.docker.internal to 192.168.2.33 in container hosts file (copy from host hosts file)
Setting gateway.docker.internal to 192.168.2.33 in container hosts file (copy from host hosts file)
Setting kubernetes.docker.internal to 127.0.0.1 in container hosts file (copy from host hosts file)
Setting host.containerhelper.internal to 172.25.224.1 in container hosts file
Starting Container
Hostname is bctest200
PublicDnsName is bctest200
Using NavUserPassword Authentication
Creating Self Signed Certificate
Self Signed Certificate Thumbprint FD50F3C12B4525E420BBBC976351BB36C2F384E6
DNS identity bctest200
Modifying Service Tier Config File with Instance Specific Settings
Modifying Service Tier Config File with settings from environment variable
Setting EnableTaskScheduler to True
Starting Service Tier
Registering event sources
Creating DotNetCore Web Server Instance
Using application pool name: BC
Using default container name: NavWebApplicationContainer
Copy files to WWW root C:\inetpub\wwwroot\BC
Create the application pool BC
Create website: NavWebApplicationContainer without SSL
Update configuration: navsettings.json
Done Configuring Web Client
Using license file 'c:\run\my\license.flf'
Import License
Creating http download site
Setting SA Password and enabling SA
Creating mt as SQL User and add to sysadmin
Creating SUPER user
Assign Premium plan for MT
Container IP Address: 172.25.229.39
Container Hostname  : bctest200
Container Dns Name  : bctest200
Web Client          : http://bctest200/BC/
Dev. Server         : http://bctest200
Dev. ServerInstance : BC
Setting bctest200 to 172.25.229.39 in host hosts file

Files:
http://bctest200:8080/ALLanguage.vsix

WARNING: You are running a container which is 118 days old.
Microsoft recommends that you always run the latest version of our containers.

Container Total Physical Memory is 31.6Gb
Container Free Physical Memory is 15.3Gb

Initialization took 126 seconds
Ready for connections!
Reading CustomSettings.config from bctest200
Creating Desktop Shortcuts for bctest200
Container bctest200 successfully created
Health check returns False, restarting container
Removing Session bctest200
bctest200
Waiting for container bctest200 to be ready

Initializing...
Setting host.docker.internal to 192.168.2.33 in container hosts file (copy from host hosts file)
Setting gateway.docker.internal to 192.168.2.33 in container hosts file (copy from host hosts file)
Setting kubernetes.docker.internal to 127.0.0.1 in container hosts file (copy from host hosts file)
Setting host.containerhelper.internal to 172.25.224.1 in container hosts file
Restarting Container
PublicDnsName unchanged
PublicDnsName is bctest200
Starting Local SQL Server
Starting Internet Information Server
Starting Service Tier
Container IP Address: 172.25.239.129
Container Hostname  : bctest200
Container Dns Name  : bctest200
Web Client          : http://bctest200/BC/
Dev. Server         : http://bctest200
Dev. ServerInstance : BC
Setting bctest200 to 172.25.239.129 in host hosts file

Files:
http://bctest200:8080/ALLanguage.vsix

Container Total Physical Memory is 31.6Gb
Container Free Physical Memory is 15.8Gb

Initialization took 33 seconds
Ready for connections!

Use:
Get-BcContainerEventLog -containerName bctest200 to retrieve a snapshot of the event log from the container
Get-BcContainerDebugInfo -containerName bctest200 to get debug information about the container
Enter-BcContainer -containerName bctest200 to open a PowerShell prompt inside the container
Remove-BcContainer -containerName bctest200 to remove the container again
docker logs bctest200 to retrieve information about URL's again
```

**Screenshots**
General tab of the event log entry:
![image](https://user-images.githubusercontent.com/12089874/177710145-5037e8ea-a055-4477-a9d6-76448ec644e6.png)

EventData in the Details-tab:
![image](https://user-images.githubusercontent.com/12089874/177710216-55d2340f-00f8-4007-a6ec-506a803c847f.png)

**Additional context**
I am using Docker on a host machine that does not have a Business Central installation, which might be the reason it's not resolving the log entries correct. 

This isn't a critical issue by any means, but wanted to let you know in case there's anything that can be done to fix it.

**Comment by freddydk** [comment]
I have noticed this as well and I don't know if anything can be done about this.

**Comment by mtaanquist** [comment]
I think I may have found a solution to this. In order to package the event log data so that it's readable without the event publisher, you need to run wevtutil with the archive-log flag. This code change worked for me:

```diff
    $eventLogName = Join-Path $eventLogFolder ($containerName + ' ' + [DateTime]::Now.ToString("yyyy-MM-dd HH.mm.ss") + ".evtx")
+   $locale = (Get-WinSystemLocale).Name
+
-   Invoke-ScriptInBcContainer -containerName $containerName -ScriptBlock { Param([string]$path, [string]$logname) 
+   Invoke-ScriptInBcContainer -containerName $containerName -ScriptBlock { Param([string]$path, [string]$logname, [string]$locale) 
        wevtutil epl $logname "$path"
+       wevtutil al "$path" /locale:$locale
-   } -ArgumentList (Get-BcContainerPath -containerName $containerName -Path $eventLogName), $logname
+   } -ArgumentList (Get-BcContainerPath -containerName $containerName -Path $eventLogName), $logname, $locale
```

I now get messages in Event Viewer with the full description:
![image](https://user-images.githubusercontent.com/12089874/177955015-05a9c0af-2b9b-434a-8e49-369518c0efea.png)

Would you like me to make a pull request for this, Freddy?

**Comment by freddydk** [comment]
Wow, thanks - yes a PR would be very welcome.

**Comment by freddydk** [comment]
Shipped in the latest prerelease

**Comment by freddydk** [comment]
Shipped in 4.0.3
